aggregators/codec.go

```diff
@@ -63,7 +63,9 @@ func (k *CombinedMetricsKey) SizeBinary() int {
 // ToProto converts CombinedMetrics to its protobuf representation.
 func (m *CombinedMetrics) ToProto() *aggregationpb.CombinedMetrics {
 	pb := aggregationpb.CombinedMetricsFromVTPool()
-	pb.ServiceMetrics = make([]*aggregationpb.KeyedServiceMetrics, 0, len(m.Services))
+	if len(m.Services) > cap(pb.ServiceMetrics) {
```

Comment:
Using `if len(m.Services) > cap(pb.ServiceMetrics)` rather than `if m.Services == nil` as @lahsivjar suggested. It is possible that in the worst case `len(m.Services) > cap(pb.ServiceMetrics)` is always true, we'll get an increasingly large Pool. We assume that go runtime will handle such growth by evicting pool entries automatically under memory stress.